sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2899,6 +2899,21 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     }
   }
 
+  /**
+   * Create a [[DescribeNamespaceStatement]].
+   *
+   * For example:
+   * {{{
+   *   DESCRIBE DATABASE [EXTENDED] database;
+   * }}}
+   */
+  override def visitDescribeNamespace(ctx: DescribeNamespaceContext): LogicalPlan =
+    withOrigin(ctx) {
+    DescribeNamespaceStatement(
+      visitMultipartIdentifier(ctx.multipartIdentifier()),
+      ctx.EXTENDED != null)
+  }
+
```

Comment:
We had better move this to line 2543 to group the related functions.
- https://github.com/apache/spark/blob/827cac1de70d233c08d7d052f59e1aa48f944e4f/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L2543